compiler/packages/babel-plugin-react-compiler/src/__tests__/fixtures/compiler/capturing-func-mutate-2.expect.md

```diff
@@ -27,7 +27,6 @@ export const FIXTURE_ENTRYPOINT = {
 import { c as _c } from "react/compiler-runtime";
 function component(a, b) {
   const $ = _c(2);
-  const y = { b };
```

Comment:
dce correctly removes non-side effectful declaration of `y`
After #31202 which cleans up non-referenced values in `HIRFunction.context`, the last lingering reference to `y` was in `LoweredFunction.dependencies` which is removed by this PR.